### PR TITLE
Do not overwrite category with blank

### PR DESCRIPTION
**Pull request by EdwardAndress**
- if the user submits the edit form with a blank option selected we should not be over-writing the respective field with an empty string
- instead, parse the params and do not update with blank values / empty strings
- there may be a neater way of doing this without an additional parse method

### CODE REVIEW

File: app/controllers/correspondence_controller.rb [hunk 1]
```diff
@@ -35,6 +35,10 @@ def search
 
   private
 
+  def parsed_edit_params
+    edit_correspondence_params.delete_if { |_key, value| value == "" }
```
Comment (jsugarman) [inline app/controllers/correspondence_controller.rb]: 👍 for delete_if and _key for unused args

File: spec/controllers/correspondence_controller_spec.rb [hunk 2]
```diff
@@ -98,6 +98,12 @@
       it 'updates the correspondence record' do
         expect(Correspondence.first.topic).to eq 'courts'
       end
+
+      it 'does not overwrite entries with blanks (if the blank dropdown option is selected)' do
+        id = all_correspondence.first.id
+        patch :update, params: { id: id, correspondence: { category: '', topic: 'courts' } }
```
Comment (jsugarman) [inline spec/controllers/correspondence_controller_spec.rb]: do u need the topic: 'courts'. would remove unless you are going to test it or is needed?!

**Comment by jsugarman** [comment]
looks good, one thing to ensure is that you are not deleting from params directly (i do not believe you are) and an alternative method would be to use except
`params.except(:courts)`

File: spec/controllers/correspondence_controller_spec.rb (hunk 2 above)
Comment (EdwardAndress) [inline spec/controllers/correspondence_controller_spec.rb]: The params will always have the :topic key, so this test is mimicking reality

**Pull request merged**